velox/cider/VeloxToCiderExpr.cpp

```
@@ -137,21 +137,26 @@ std::shared_ptr<Analyzer::Expr> VeloxToCiderExprConverter::wrap_expr_with_cast(
 std::shared_ptr<Analyzer::Expr> VeloxToCiderExprConverter::toCiderExpr(
     std::shared_ptr<const ITypedExpr> v_expr,
     std::unordered_map<std::string, int> col_info) const {
-  if (auto constant =
-          std::dynamic_pointer_cast<const ConstantTypedExpr>(v_expr)) {
-    return toCiderExpr(constant);
-  }
-  if (auto field_expr =
-          std::dynamic_pointer_cast<const FieldAccessTypedExpr>(v_expr)) {
-    return toCiderExpr(field_expr, col_info);
-  }
-  if (auto call_expr = std::dynamic_pointer_cast<const CallTypedExpr>(v_expr)) {
-    return toCiderExpr(call_expr, col_info);
-  }
-  if (auto cast_expr = std::dynamic_pointer_cast<const CastTypedExpr>(v_expr)) {
-    return toCiderExpr(cast_expr, col_info);
+  try {
+    if (auto constant =
+            std::dynamic_pointer_cast<const ConstantTypedExpr>(v_expr)) {
+      return toCiderExpr(constant);
+    }
+    if (auto field_expr =
+            std::dynamic_pointer_cast<const FieldAccessTypedExpr>(v_expr)) {
+      return toCiderExpr(field_expr, col_info);
+    }
+    if (auto call_expr = std::dynamic_pointer_cast<const CallTypedExpr>(v_expr)) {
+      return toCiderExpr(call_expr, col_info);
+    }
+    if (auto cast_expr = std::dynamic_pointer_cast<const CastTypedExpr>(v_expr)) {
+      return toCiderExpr(cast_expr, col_info);
+    }
+    return nullptr;
```

Should we just throw an exception here? In the end, we should support all 6 implements for ```ITypedExpr```?

---

at expression translation level, we return nullptr if it fails and will detect this in node level translation and then throw exception with node info provided. Eventually, we will support all implements of ITypedExpr to more detailed Omnisci exprs.